src/test/stateful_config_test.cpp

```diff
@@ -0,0 +1,177 @@
+
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "../modelconfig.hpp"
+#include "../modelinstance.hpp"
+#include "test_utils.hpp"
+
+using namespace ovms;
+
+using testing::_;
+using testing::Return;
+
+static const char* modelDefaultConfig = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy",
+                "target_device": "CPU",
+                "model_version_policy": {"latest": {"num_versions":1}},
+                "nireq": 100,
+                "shape": {"b": "(1,10) "}
+            }
+        }
+    ],
+    "pipeline_config_list": [
+        {
+            "name": "pipeline1Dummy",
+            "inputs": ["custom_dummy_input"],
+            "nodes": [
+                {
+                    "name": "dummyNode",
+                    "model_name": "dummy",
+                    "type": "DL model",
+                    "inputs": [
+                        {"b": {"node_name": "request",
+                               "data_item": "custom_dummy_input"}}
+                    ],
+                    "outputs": [
+                        {"data_item": "a",
+                         "alias": "new_dummy_output"}
+                    ]
+                }
+            ],
+            "outputs": [
+                {"custom_dummy_output": {"node_name": "dummyNode",
+                                         "data_item": "new_dummy_output"}
+                }
+            ]
+        }
+    ]
+})";
+
+static const char* modelStatefulChangedConfig = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy",
+                "target_device": "CPU",
+                "model_version_policy": {"latest": {"num_versions":1}},
+                "nireq": 100,
+                "stateful": true,
+                "low_latency_transformation": true,
+                "sequence_timeout": 120,
+                "max_sequence_number": 1000,
```

Comment:
why do we need to limit the sequence length?

Reply:
The stateful pipeline parameter setting was added in the design so that user has the possibility to control sequences.

Reply:
It's actually not sequence length limit. Sequence can have unlimited amount of requests. If user doesn't send SEQUENCE_END signal the sequence can potentially be infinite. 

**max_sequence_number** parameter purpose is to **limit the number of sequences handled by the model instance concurrently**. As handling each sequence (holding its state or whole infer request) requires resources, setting this parameter can limit number of sequences that can be processed concurrently by the model, so it doesn't consume too much memory.